tests/function-controller/pkg/apirule/apirule.go

```diff
@@ -85,3 +91,73 @@ func (a *APIRule) Delete() error {
 
 	return nil
 }
+
+func (a *APIRule) get() (*apiruleTypes.APIRule, error) {
+	u, err := a.resCli.Get(a.name)
+	if err != nil {
+		return &apiruleTypes.APIRule{}, errors.Wrapf(err, "while getting ApiRule %s in namespace %s", a.name, a.namespace)
+	}
+
+	apirule := &apiruleTypes.APIRule{}
+	err = runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, apirule)
+	if err != nil {
+		return &apiruleTypes.APIRule{}, err
+	}
+
+	return apirule, nil
+}
+
+func (a *APIRule) WaitForStatusRunning() error {
+	apirule, err := a.get()
+	if err != nil {
+		return err
+	}
+
+	if isStateReady(apirule.Status) {
+		a.log.Logf("%s is ready", a.name)
+		return nil
+	}
+	a.log.Logf("ApiRule %s is not ready", a.name)
+
+
+	ctx, cancel := context.WithTimeout(context.Background(), a.waitTimeout)
+	defer cancel()
+
+	condition := a.isApiRuleReady(a.name)
+	_, err = watchtools.Until(ctx, apirule.GetResourceVersion(), a.resCli.ResCli, condition)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (a *APIRule) isApiRuleReady(name string) func(event watch.Event) (bool, error) {
+	return func(event watch.Event) (bool, error) {
+		u, ok := event.Object.(*unstructured.Unstructured)
+		if !ok {
+			return false, shared.ErrInvalidDataType
+		}
+		if u.GetName() != name {
+			a.log.Logf("names mismatch, object's name %s, supplied %s", u.GetName(), name)
+			return false, nil
+		}
+
+		apirule := apiruleTypes.APIRule{}
+		err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, &apirule)
+		if err != nil {
+			return false, err
+		}
+
+		if isStateReady(apirule.Status) {
+			a.log.Logf("Apirule %s is ready:\n%v", name, u)
+			return true, nil
+		}
+
+		a.log.Logf("Apirule %s is not ready", name)
+		return false, nil
+	}
+}
+
+func isStateReady(a apiruleTypes.APIRuleStatus) bool {
```

Comment:
If I wanted to do func (a *APIRule) then I'd have to do a get for function here, because APIRule is
```
type APIRule struct {
	resCli      *resource.Resource
	name        string
	namespace   string
	waitTimeout time.Duration
	log         shared.Logger
}
```

and I can't create methods for the types I do not own

Reply:
I mean about `func (a *APIRule) isStateReady(a apiruleTypes.APIRuleStatus) bool {}`, not to changing args of function.